src/System.Private.CoreLib/src/System/String.Searching.cs

```diff
@@ -174,22 +175,37 @@ private unsafe int IndexOfAny(char value1, char value2, char value3, int startIn
         private unsafe int IndexOfCharArray(char[] anyOf, int startIndex, int count)
         {
             // use probabilistic map, see InitializeProbabilisticMap
-            uint* charMap = stackalloc uint[PROBABILISTICMAP_SIZE];
+            ProbabilisticMap map = default(ProbabilisticMap);
+            uint* charMap = (uint*)&map;
+
             InitializeProbabilisticMap(charMap, anyOf);
 
+            return IndexOfCharArrayFilter(anyOf, startIndex, charMap, count);
+        }
+
+        private unsafe int IndexOfCharArrayFilter(char[] anyOf, int startIndex, uint* charMap, int count)
```

Comment:
Is it still necessary/beneficial to have `IndexOfCharArrayFilter` factored out into separate method?

Reply:
Nope, better without that now. Re-joined.